### PR TITLE
Feature location filter

### PR DESCRIPTION
**Pull request by jodelajo**
## What this feature does
- Added location searchbar
- Added autocomplete datalist
- Added filter
- Fixed style issues companycard

Schermafbeelding 2022-01-24 om 13 08 44

Schermafbeelding 2022-01-24 om 13 08 20

### CODE REVIEW

File: src/Pages/HomePage.css [hunk 1]
```diff
@@ -4,14 +4,23 @@
...
-  text-align: center;
+.general {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  width: 100vw;
+  position: absolute;
 }
+/* .general h1 {
```
Comment (Reinoptland) [inline src/Pages/HomePage.css]: Try to remove any commented code before you PR

File: src/Pages/HomePage.js [hunk 2]
```diff
@@ -21,14 +23,28 @@ export default function HomePage() {
...
-        {agencies.map((agency) => {
-          return (
-            <div className="block">
-              <Agencies agency={agency} />
-            </div>
-          );
-        })}
+        {agencies
+          .filter((loc) => {
+            if (searchInput === "") {
```
Comment (Reinoptland) [inline src/Pages/HomePage.js]:
```javascript
if (searchInput === "" ||  loc.city.toLowerCase().includes(searchInput.toLocaleLowerCase())) {
  return true
} else {
  return false
}
```

or 
```javascript
.filter((loc) => searchInput === "" ||  loc.city.toLowerCase().includes(searchInput.toLocaleLowerCase()))
```

or 
```javascript
// somewhere in component 
const searchInputEmpty = searchInput === ""
const agencyMatchesSearchQuery = (searchInput, agency) =>  agency.city.toLowerCase().includes(searchInput.toLocaleLowerCase())

.filter((loc) => searchInputEmpty  ||  agencyMatchesSearchQuery(searchInput, loc))
```

File: src/components/Agencies/Agencies.css [hunk 3]
```diff
@@ -42,14 +38,15 @@
 }
 .company-info h5 {
   margin-bottom: 5px;
+  overflow-wrap: break-word;
```
Comment (Reinoptland) [inline src/components/Agencies/Agencies.css:41]: This already fixes a lot

We need to dive deeper into the css of the card as a whole to fix it more fundementally

File: src/components/Agencies/Agencies.js [hunk 4]
```diff
@@ -15,7 +14,7 @@ export default function Agencies({ agency }) {
         </div>
         <div className="agency-card-info" key={agency.id}>
           <div className="company-name">
-            <h2>{agency.name}</h2>
+            <h2>{agency.name.replaceAll("&amp;", "&")}</h2>
```
Comment (Reinoptland) [inline src/components/Agencies/Agencies.js:17]: Nice!

File: src/components/LocationSearch/LocationSearch.css [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  padding-left: 6px;
+}
+.resetButton {
+  padding: 0 10px;
+  color: white;
+  background-color: rgb(8, 8, 133);
+  border: none;
+  margin-left: 8px;
+  border-radius: 4px;
+  -webkit-border-radius: 4px;
```
Comment (Reinoptland) [inline src/components/LocationSearch/LocationSearch.css]: Autoprefixer will take care of this automatically

File: src/components/LocationSearch/LocationSearch.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+          throw Error("Data ophalen is mislukt");
+        }
+        return res.json();
+      })
+      .then((data) => {
+        setLocations(data);
+        setError(null);
+      })
+      .catch((err) => {
+        setError(err.message);
```
Comment (Reinoptland) [inline src/components/LocationSearch/LocationSearch.js:26]: hey, nice - error handling

File: src/components/LocationSearch/LocationSearch.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+  }, []);
+
+  const locationsArray =
+    locations &&
+    Object.keys(locations).map((name) => {
+      const obj = {};
+      obj[name] = locations[name];
+      return obj;
+    });
+  console.log(locationsArray);
```
Comment (Reinoptland) [inline src/components/LocationSearch/LocationSearch.js]: No logs pls

File: src/components/LocationSearch/LocationSearch.js [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+        setError(null);
+      })
+      .catch((err) => {
+        setError(err.message);
+      });
+  }, []);
+
+  const locationsArray =
+    locations &&
+    Object.keys(locations).map((name) => {
```
Comment (Reinoptland) [inline src/components/LocationSearch/LocationSearch.js]: Maybe turn it into a function like: 

```javascript
convertCityAgencyCountObjectToArray(locations){
  return Object.keys(locations).map((name) => { ... etc
}
const locationsArray = convertCityAgencyCountObjectToArray(locations)
```

File: src/components/LocationSearch/LocationSearch.js [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+      const obj = {};
+      obj[name] = locations[name];
+      return obj;
+    });
+  console.log(locationsArray);
+
+  function refreshPage() {
+    window.location.reload(false);
```
Comment (Reinoptland) [inline src/components/LocationSearch/LocationSearch.js]: Consider using: searchLocation("") 

No reload required!

**Review by Reinoptland — CHANGES_REQUESTED** [review]

File: src/components/LocationSearch/LocationSearch.js (hunk 9 above)
Comment (jodelajo) [reply src/components/LocationSearch/LocationSearch.js]: check

File: src/components/LocationSearch/LocationSearch.js (hunk 7 above)
Comment (jodelajo) [reply src/components/LocationSearch/LocationSearch.js]: check

File: src/components/LocationSearch/LocationSearch.css (hunk 5 above)
Comment (jodelajo) [reply src/components/LocationSearch/LocationSearch.css]: check

**Review by Reinoptland — APPROVED** [review]

**Pull request merged**